ledger/participant-integration-api/src/main/scala/platform/usermanagement/PersistentUserManagementStore.scala

```diff
@@ -0,0 +1,259 @@
+  private val logger = ContextualizedLogger.get(this.getClass)
+
+  implicit private val loggingContext: LoggingContext = LoggingContext.newLoggingContext(identity)
+
+  // TODO participant user management: This is probably not the best place/time to create admin user
+  if (createAdminUser)
+    createUser(
+      InMemoryUserManagementStore.AdminUser.user,
+      InMemoryUserManagementStore.AdminUser.rights,
+    )
+
+  override def createUser(
+      user: domain.User,
+      rights: Set[domain.UserRight],
+  ): Future[Result[Unit]] = {
+    inTransaction { implicit connection: Connection =>
+      withoutUser(user.id) {
+        if (rights.size > maxNumberOfUserRightsPerUserLimit) {
+          throw TooManyUserRightsRuntimeException(user.id)
+        } else {
+          val nowMicros = epochMicroSeconds()
+          val internalId = backend.createUser(user, createdAt = nowMicros)(connection)
+          rights.foreach(right =>
+            backend.addUserRight(internalId = internalId, right = right, grantedAt = nowMicros)(
+              connection
+            )
+          )
+          if (backend.countUserRights(internalId)(connection) > maxNumberOfUserRightsPerUserLimit) {
+            throw TooManyUserRightsRuntimeException(user.id)
+          } else {
+            ()
+          }
+        }
+      }
+    }.map(tapSuccess { _ =>
+      // // TODO participant user management: Unit test logged messages
+      logger.info(s"Created new user: ${user}")
+      foreachEachUserRight(rights) { suffix =>
+        logger.info(s"New user: ${user} was created with right $suffix")
+      }
+    })(scala.concurrent.ExecutionContext.parasitic)
+  }
+
+  override def deleteUser(id: UserId): Future[Result[Unit]] = {
+    inTransaction { implicit connection =>
+      if (!backend.deleteUser(id = id)(connection = connection)) {
+        Left(UserNotFound(userId = id))
+      } else {
+        Right(())
+      }
+    }.map(tapSuccess { _ =>
+      logger.info(s"Deleted user with id: ${id}")
+    })(scala.concurrent.ExecutionContext.parasitic)
+  }
+
+  override def getUser(id: UserId): Future[Result[domain.User]] = {
+    inTransaction { implicit connection =>
+      withUser(id) { dbUser =>
+        dbUser.domainUser
+      }
+    }
+  }
+
+  override def grantRights(
+      id: UserId,
+      rights: Set[domain.UserRight],
+  ): Future[Result[Set[domain.UserRight]]] = {
+    inTransaction { implicit connection =>
+      withUser(id = id) { user =>
+        if (rights.size > maxNumberOfUserRightsPerUserLimit) {
+          throw TooManyUserRightsRuntimeException(user.domainUser.id)
+        } else {
+          val addedRights = rights.filter { right =>
+            if (!backend.userRightExists(user.internalId, right)(connection)) {
+              backend.addUserRight(
+                internalId = user.internalId,
+                right = right,
+                grantedAt = epochMicroSeconds(),
```

Comment:
Let the DB server fill this field using a default.

Reply:
This way all measurements of `created_at` and `granted_at` are done using the same clock.

Reply:
> Let the DB server fill this field using a default.
Ok

> This way all measurements of created_at and granted_at are done using the same clock.
Currently they use the same clock as well (i.e. whatever JVM clock mechanism `epochMicroSeconds()` uses) or am I missing something?